src/exporters/vhi.py

```diff
@@ -0,0 +1,172 @@
+"""
+functionality:
+1) download VHI data in parallel
+2)
+
+CDS functions:
+- create_area()
+- _filename_from_selection_request()
+- make_filename()
+- _print_api_request()
+- _export()
+- get_era5_times()
+- get_dataset()
+- _correct_input()
+- _check_iterable()
+- create_selection_request()
+- export()
+"""
+
+from pathlib import Path
+from typing import Dict, Optional, List
+from ftplib import FTP
+import multiprocessing
+from pprint import pprint
+import re
+
+# unnecessary extra dependency?
+from pathos.multiprocessing import ProcessingPool as Pool
+
+from .base import BaseExporter, Region
+
+# TODO: separate into general ftp exporter class?
+# class FTPExporter():
+#     """"""
+
+class VHIExporter(BaseExporter):
+    """Exports Vegetation Health Index from NASA site
+
+    ftp.star.nesdis.noaa.gov
+    """
+
+    def __init__(self, data_folder: Path = Path('data')) -> None:
+        super().__init__(data_folder)
+
+    def get_ftp_filenames() -> List:
+        """  get the filenames containing VHI """
+        with FTP('ftp.star.nesdis.noaa.gov') as ftp:
+            ftp.login()
+            ftp.cwd('/pub/corp/scsb/wguo/data/Blended_VH_4km/VH/')
+
+            # append the filenames to a list
+            listing = []
+            ftp.retrlines("LIST", listing.append)
+            # extract the filename
+            filepaths = [f.split(' ')[-1] for f in listing]
+            # extract only the filenames of interest
+            vhi_files = [f for f in filepaths if ".VH.nc" in f]
+
+        return vhi_files
+
+    @staticmethod
+    def chunks(l, n):
```

Comment:
@gabrieltseng this function appears to work (no problems my side with the code). I am not *certain* that it's parallelized but i'm 90% sure because the order that the files return is not sequential and the list is sequential when you call the FTP server. So I'm pretty sure it's working as expected.

BUT it's failing the mypy checks:
```
src/exporters/vhi.py:152: error: Argument 1 to "len" has incompatible type "int"; expected "Sized"
src/exporters/vhi.py:154: error: Value of type "int" is not indexable
```

Reply:
Oh it's all fixed by changing to : 
`def chunks(self, l: List, n: int) -> Generator:`